[SelectionDAG] Change computeAliasing signature from optional<uint64> to LocationSize.

davemgreen opened the pull request:
This is another smaller step of #70452, changing the signature of computeAliasing() from optional<uint64_t> to LocationSize, and follow-up changes in DAGCombiner::mayAlias(). There are some test change due to the previous AA->isNoAlias call incorrectly using an unknown size (~UINT64_T(0)). This should then be improved again in #70452 when the types are known to be scalable.

RKSimon commented on llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:
```diff
@@ -27835,40 +27835,42 @@ bool DAGCombiner::mayAlias(SDNode *Op0, SDNode *Op1) const {
+        Offset = (LSN->getAddressingMode() == ISD::PRE_INC) ? C->getSExtValue()
+                 : (LSN->getAddressingMode() == ISD::PRE_DEC)
+                     ? -1 * C->getSExtValue()
+                     : 0;
```
separate clang-format cleanup?

davemgreen committed:
[SelectionDAG] Change computeAliasing signature from optional<uint64> to LocationSize.

This is another smaller step of #70452, changing the signature of
computeAliasing() from optional<uint64_t> to LocationSize, and follow-up
changes in DAGCombiner::mayAlias(). There are some test change due to the
previous AA->isNoAlias call incorrectly using an unknown size (~UINT64_T(0)).
This should then be improved again in #70452 when the types are known to be
scalable.

lukel97 commented:
Thanks for landing this, this [fixes a miscompile](https://github.com/llvm/llvm-project/commit/63725ab1196ac50509ad382fc12c56f6d8b5d874) we were seeing on RISC-V on SPEC CPU 2017 with a specific vector configuration `-mrvv-vector-bits=zvl`.

Would you have any objections to backporting this to 18.x?

davemgreen commented:
Hi. Yeah that sounds good. I hadn't realized this came up in practice.

Did you want to create the ticket and whatnot?

lukel97 commented:
/cherry-pick ea5aad7ca37e557e7d6cf0746c9f94f7fe3ce28c

lukel97 commented:
/cherry-pick 6e41d60

lukel97 commented:
/cherry-pick 6e41d60a717132fadac74abe61ac6a9b1ca98778

(Sorry for the noise, GitHub seems to need more than 7 characters for this sha)

lukel97 commented:
/cherry-pick 6e41d60a717132fadac74abe61ac6a9b1ca98778 63725ab1196ac50509ad382fc12c56f6d8b5d874
